llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -1809,6 +1809,7 @@ Instruction *InstCombinerImpl::foldOpIntoPhi(Instruction &I, PHINode *PN) {
   // Check to see whether the instruction can be folded into each phi operand.
   // If there is one operand that does not fold, remember the BB it is in.
   SmallVector<Value *> NewPhiValues;
+  SmallVector<unsigned int> OpsToMoveUseTo;
   BasicBlock *NonSimplifiedBB = nullptr;
   Value *NonSimplifiedInVal = nullptr;
```

Comment:
It looks like the logic is now duplicated between OpsToMoveUseTo and NonSimplifiedBB/InVal? Both are non-simplified cases where we move the operation to the predecessors, so we shouldn't need separate handling for the case where there is one and the case where there are multiple.

Reply:
Fixed, thanks